assets/app/view/game/track_conversion.rb

```diff
@@ -0,0 +1,170 @@
+          store(:routes, @routes, skip: true)
+          store(:selected_route, route, skip: true)
+        end
+
+        @routes.each(&:clear_cache!)
+
+        trains = trains.flat_map do |train|
+          onclick = lambda do
+            unless (route = @routes.find { |t| t.train == train })
+              route = Engine::Route.new(@game, @game.phase, train, abilities: @abilities, routes: @routes)
+              @routes << route
+              store(:routes, @routes)
+            end
+            store(:selected_route, route)
+          end
+
+          selected = @selected_route&.train == train
+
+          style = {
+            border: "solid 3px #{selected ? color_for(:font) : color_for(:bg)}",
+            display: 'inline-block',
+            cursor: selected ? 'default' : 'pointer',
+            margin: '0.1rem 0rem',
+            padding: '3px 6px',
+            minWidth: '1.5rem',
+            textAlign: 'center',
+            whiteSpace: 'nowrap',
+          }
+
+          route = active_routes.find { |t| t.train == train }
+          children = []
+          if route
+            _revenue, invalid = begin
+              # need this call to force errors
+              [route.revenue.to_s, nil]
+            rescue Engine::GameError => e
+              ['N/A', e.to_s]
+            end
+
+            bg_color = route_prop(@routes.index(route), :color)
+            style[:backgroundColor] = bg_color
+            style[:color] = contrast_on(bg_color)
+
+            td_props = { style: { paddingRight: '0.8rem' } }
+
+            children << h('td.right.middle', td_props, "#{route.distance_str} Stops")
+            children << h(:td, route.revenue_str)
+          elsif !selected
+            style[:border] = '1px solid'
+            style[:padding] = '5px 8px'
+          end
+
+          invalid_props = {
+            attrs: {
+              colspan: '4',
+            },
+            style: {
+              padding: '0 0 0.4rem 0.4rem',
+            },
+          }
+          train_name = 'Segment:'
+          [
+            h(:tr, [h('td.middle', [h(:div, { style: style, on: { click: onclick } }, train_name)]), *children]),
+            invalid ? h(:tr, [h(:td, invalid_props, invalid)]) : '',
+          ]
+        end
+
+        div_props = {
+          key: 'route_selector',
+          hook: {
+            destroy: -> { cleanup },
+          },
+        }
+        table_props = {
+          style: {
+            marginTop: '0.5rem',
+            textAlign: 'left',
+          },
+        }
+
+        instructions = 'Click revenue centers, again to cycle paths. '\
+          'Must be from city/offboard to city/offboard'
+        h3_text = 'Select Segment for Conversion'
+
+        h(:div, div_props, [
+          h(:h3, h3_text),
+          h('div.small_font', instructions),
+          train_help,
+          h(:table, table_props, [
+            h(:tbody, trains),
+          ]),
+          actions,
+        ].compact)
```

Comment:
why do you need compact here

Reply:
No reason. Just left over from leveraged code.